datasets/metooma/metooma.py

```diff
@@ -0,0 +1,130 @@
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""TODO: Add a description here."""
+
+from __future__ import absolute_import, division, print_function
+
+import csv
+
+import datasets
+
+
+_CITATION = """\
+@inproceedings{gautam2020metooma,
+    title={# MeTooMA: Multi-Aspect Annotations of Tweets Related to the MeToo Movement},
+    author={Gautam, Akash and Mathur, Puneet and Gosangi, Rakesh and Mahata, Debanjan and Sawhney, Ramit and Shah, Rajiv Ratn},
+    booktitle={Proceedings of the International AAAI Conference on Web and Social Media},
+    volume={14},
+    pages={209--216},
+    year={2020} }
+"""
+
+_DESCRIPTION = """\
+The dataset consists of tweets belonging to #MeToo movement on Twitter, labelled into different categories.
+Due to Twitter's development policies, we only provide the tweet ID's and corresponding labels,
+other data can be fetched via Twitter API.
+The data has been labelled by experts, with the majority taken into the account for deciding the final label.
+We provide these labels for each of the tweets. The labels provided for each data point
+includes -- Relevance, Directed Hate, Generalized Hate,
+Sarcasm, Allegation, Justification, Refutation, Support, Oppose
+"""
+
+
+_TRAIN_DOWNLOAD_URL = "https://raw.githubusercontent.com/akash418/public-data-repo/main/MeTooMMD_train.csv"
+_TEST_DOWNLOAD_URL = "https://raw.githubusercontent.com/akash418/public-data-repo/main/MeTooMMD_test.csv"
+
+
+class MeTooMA(datasets.GeneratorBasedBuilder):
+    """MeTooMA dataset -- Dataset providing labeled information for tweets belonging to the MeToo movement"""
+
+    VERSION = datasets.Version("1.1.0")
+
+    def _info(self):
+        # This method pecifies the datasets.DatasetInfo object which contains informations and typings for the dataset
+
+        return datasets.DatasetInfo(
+            # This is the description that will appear on the datasets page.
+            description=_DESCRIPTION,
+            features=datasets.Features(
+                {
+                    "TweetId": datasets.Value("string"),
+                    "Text_Only_Informative": datasets.Value("string"),
+                    "Image_Only_Informative": datasets.Value("string"),
+                    "Directed_Hate": datasets.Value("string"),
+                    "Generalized_Hate": datasets.Value("string"),
+                    "Sarcasm": datasets.Value("string"),
+                    "Allegation": datasets.Value("string"),
+                    "Justification": datasets.Value("string"),
+                    "Refutation": datasets.Value("string"),
+                    "Support": datasets.Value("string"),
+                    "Oppose": datasets.Value("string"),
```

Comment:
Could you use the `datasets.ClassLabel(names=["neg", "pos"])` feature type for the different labels please ? (and replace neg/pos by the expected strings if necessary)

Reply:
Resolved in the latest commit.